apps/openmw/mwgui/spellmodel.cpp

```diff
@@ -148,6 +148,17 @@ namespace MWGui
         return mSpells.size();
     }
 
+    SpellModel::ModelIndex SpellModel::getSelectedIndex() const
+    {
+        ModelIndex selected = -1;
+        for (SpellModel::ModelIndex i = 0; i<int(getItemCount()); ++i)
+        {
+            if (getItem(i).mSelected)
+                selected = i;
```

Comment:
I suppose there should be "break". Or maybe you can just store selected index in the variable and update it in the SpellModel::update().

Reply:
I've added a break